lyft/envoy
# test: various test changes to support OS X builds

test/common/network: 
- make fewer assumptions about event timing and sequence
- use the offset of `sun_path` rather than assuming the size and number of preceding sockaddr fields
- OS X treats ::/96 IPv6 addresses as IPv6-mapped IPv4 addresses (per a deprecated RFC); use a different address range so expected strings match across OS type

test/common/http: use a more recent date for log tests to avoid an OS X time_t limitation: struct tm before 1901-12-13 20:45:52 cannot be converted to time_t.

test/common/runtime: handle difference between cp on Linux and OS X

test/common/upstream: disable ring_hash_lb_test on OS X

test/exe: OS X reports abort signals slightly differently than Linux

test/test_common: remove getSomeLoopbackAddress (OS X does not, by default, support IPv4 loopback other than 127.0.0.1) since all tests bind port 0

(Split out from #1348, in support of #128.)
